includes/object-cache.php

```diff
@@ -1090,7 +1088,9 @@ public function fetch_info() {
         }
 
         if ( defined( 'WP_REDIS_CLUSTER' ) ) {
-            $info = $this->redis->info( current( array_values( WP_REDIS_CLUSTER ) ) );
+            $info = ($this->determine_client() === 'predis')
+                ? $this->redis->getClientFor( $this->build_connection_id() )->info()
+                : $this->redis->info( $this->build_connection_id() );
```

Comment:
Make sure this works with `Credis` as well 😬

Reply:
https://github.com/rhubarbgroup/redis-cache/wiki

Not sure why, but we also support Credis.

```php
// supported clients: `phpredis`, `credis`, `predis` and `hhvm`
define( 'WP_REDIS_CLIENT', 'credis' );
```

Reply:
I know, but I have no idea what is `Credis` nor how to install it locally, let me check and see what can I achieve.

Reply:
@tillkruss do we really need to support `Credis`? last update to their code was in 2010 https://github.com/octo/credis (https://code.google.com/archive/p/credis/) .. do people still use it?

Reply:
Before I took over the Predis repo 🙃

The Credis support is fairly limited.

Reply:
after some time trying to debug and see if Credis will work without any problems, I found out that Credis does not handle `MOVED` or `ASK` errors instead it will throw an error 
![image](https://user-images.githubusercontent.com/27624/177771841-7b713a74-d548-4277-ba06-424cee4faab7.png)

so either we will have to handle it from our end, or we have to wait for the library to support them.